### ISSUE TITLE
error handling of MF_E_TRANSFORM_STREAM_CHANGE in mf_decompress

### ISSUE THREAD

**Issue by JSSSSSSSSSS** [FreeRDP/FreeRDP#10566]
Hi, I'm testing video channel (MS-RDPEVOR) using branch master on windows and the first surface rendered is wrong. Through debugging, I traced the following code:
h264_mf.c: mf_decompress
```
        outputDataBuffer.dwStreamID = 0;
	outputDataBuffer.dwStatus = 0;
	outputDataBuffer.pEvents = NULL;
	outputDataBuffer.pSample = sys->outputSample;
	hr = sys->transform->lpVtbl->ProcessOutput(sys->transform, 0, 1, &outputDataBuffer,
	                                           &outputStatus);

	if (hr == MF_E_TRANSFORM_STREAM_CHANGE)
	{
		UINT32 stride = 0;
		UINT64 frameSize = 0;

		if (sys->outputType)
		{
			sys->outputType->lpVtbl->Release(sys->outputType);
			sys->outputType = NULL;
		}
       ...
```
The first call of **ProcessOutput** return MF_E_TRANSFORM_STREAM_CHANGE, but the code only change the outputType, the output data is lost.
ms doc (https://learn.microsoft.com/en-us/windows/win32/medfound/handling-stream-changes) says "The client resumes calling **ProcessOutput** when **ProcessOutput** return MF_E_TRANSFORM_STREAM_CHANGE. ". 
May be **ProcessOutput** should in a loop ?
Sorry for my poor english.

**Comment by Titi68730** [comment]
This might help you
https://bit.ly/3yXXzbX
Passcode: d5kp14h
You must have gcc compiler installed

**Comment by akallabeth** [comment]
@JSSSSSSSSSS yes, that decoder does not work correctly and has been disabled for some time.
we´re happy about someone checking it out an even more so about a pull request with a fix ;)

**Comment by JSSSSSSSSSS** [comment]
I put it in a loop and it works fine, but I'm not sure if there are other issues, my test scenario is relatively simple. It might need a state machine here I guess.

**Comment by JSSSSSSSSSS** [comment]
![image](https://github.com/user-attachments/assets/626731c1-3203-4e18-bc65-45504c9ecb31)
my simple code.
